gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/grouping/functions/BuildBase.java

```diff
@@ -263,20 +315,39 @@ protected void setAggregateValues(EPGMAttributed element) {
    * Sets the given property values as new properties at the given element.
    *
    * @param element attributed element
-   * @param values  aggregate values
+   * @param label element label
+   * @param values aggregate values
    */
   protected void setAggregateValues(
-    EPGMAttributed element, PropertyValueList values) {
+    EPGMAttributed element, String label, PropertyValueList values) {
     if (doAggregate()) {
-      Iterator<PropertyValueAggregator> aggIt = valueAggregators.iterator();
       Iterator<PropertyValue> valueIt = values.iterator();
 
-      while (aggIt.hasNext() && valueIt.hasNext()) {
-        PropertyValueAggregator aggregator = aggIt.next();
-        PropertyValue value = valueIt.next();
-
-        element.setProperty(aggregator.getAggregatePropertyKey(), value);
+      for (PropertyValueAggregator valueAggregator : valueAggregators) {
+        // check if the aggregator is used for the elements label
+        if (verifyAggregatorForLabel(valueAggregator.getAggregatePropertyKey(), label) ||
+          // or used for all vertices
+          verifyAggregatorForLabel(
+            valueAggregator.getAggregatePropertyKey(), Grouping.VERTEX_AGGREGATOR)) {
+          PropertyValue value = valueIt.next();
+          element.setProperty(
+            valueAggregator.getAggregatePropertyKey(),
+            value);
+        }
       }
     }
   }
+
+  /**
+   * Verifies if the given aggregator property key shall be used fo the elements label.
+   *
+   * @param aggregatorPropertyKey aggregator property key
+   * @param elementLabel element label
+   * @return true if the aggregator shall be used for the element
+   */
+  private boolean verifyAggregatorForLabel(String aggregatorPropertyKey, String elementLabel) {
+    return labelWithAggregatorPropertyKeys
+      .getOrDefault(aggregatorPropertyKey, Sets.newHashSet())
```

Comment:
consider how often this is called. creating a new hash set each time a property key is not in the map is not feasible.